### PR TITLE
skip private address when creating routes

### PR DESCRIPTION
**Pull request by mattkasun**
## Describe your changes
skip creating explicit endpoint/server routes to private address

## Provide testing steps
see GRA-1551

## Checklist before requesting a review
- [x] My changes affect only 10 files or less.
- [x] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [x] If it is a bugfix, my code is <= 200 lines.
- [ ] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [ ] Netclient & Netmaker are awesome.

### CODE REVIEW

**Review by abhishek9686 — APPROVED** [review]

File: routes/routes_linux.go [hunk 1]
```diff
@@ -81,6 +87,12 @@ func SetNetmakerPeerEndpointRoutes(defaultInterface string) error {
 	currentPeers := config.GetHostPeerList()
 	for i := range currentPeers {
 		peer := currentPeers[i]
+		if peer.Endpoint == nil {
```
Comment (gabrielseibel1) [inline routes/routes_linux.go:90]: Can `peer.Endpoint != nil` but `peer.Endpoint.IP == nil`?

File: local/local.go [hunk 2]
```diff
@@ -36,14 +36,13 @@ func SetIPForwarding() error {
 
 // SetIPForwardingUnix - sets the ipforwarding for linux
 func SetIPForwardingUnix() error {
-	x := []byte{0x01}
 	// ipv4
-	if err := os.WriteFile("/proc/sys/net/ipv4/ip_foward", x, os.ModePerm); err != nil {
+	if err := os.WriteFile("/proc/sys/net/ipv4/ip_forward", []byte("1"), os.ModePerm); err != nil {
```
Comment (gabrielseibel1) [inline local/local.go:40]: This change effectively makes `x := [49]` instead of `x := [1]`. Is this intended, and why?

File: local/local.go [hunk 3]
```diff
@@ -36,14 +36,13 @@ func SetIPForwarding() error {
...
 		logger.Log(0, "WARNING: Error encountered setting ip forwarding. This can break functionality.")
 		return err
 	}
 	// ipv6
-	if err := os.WriteFile("/proc/sys/net/ipv6/conf/all/foward", x, os.ModePerm); err != nil {
+	if err := os.WriteFile("/proc/sys/net/ipv6/conf/all/forward", []byte("1"), os.ModePerm); err != nil {
```
Comment (gabrielseibel1) [inline local/local.go:45]: Same as [above](https://github.com/gravitl/netclient/pull/360/files#r1174076443)

**Review by gabrielseibel1 — CHANGES_REQUESTED** [review]
Looks good, left some appointments in comments but overall looks ok :)

File: local/local.go (hunk 2 above)
Comment (mattkasun) [reply local/local.go:40]: The other way give error Invalid Arg, the new way results in file containing 1

File: routes/routes_linux.go (hunk 1 above)
Comment (mattkasun) [reply routes/routes_linux.go:90]: perhaps, I will change

**Review by theguy951357 — APPROVED** [review]
pr steps pass
critical list passes

File: routes/routes_linux.go [hunk 4]
```diff
@@ -81,6 +87,12 @@ func SetNetmakerPeerEndpointRoutes(defaultInterface string) error {
 	currentPeers := config.GetHostPeerList()
 	for i := range currentPeers {
 		peer := currentPeers[i]
+		if peer.Endpoint.IP == nil {
```
Comment (abhishek9686) [inline routes/routes_linux.go:90]: this check will cause a nil pointer deference error if `peer.Endpoint is nil`

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

File: routes/routes_linux.go (hunk 4 above)
Comment (mattkasun) [reply routes/routes_linux.go]: changed back to if peer.Endpoint == nil

**Review by abhishek9686 — APPROVED** [review]

**Pull request merged**